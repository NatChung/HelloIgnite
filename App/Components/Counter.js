import React, { Component } from 'react'
import PropTypes from 'prop-types';
import styles from './Styles/CounterStyle'
import {
  TouchableOpacity,
  Text,
  View
} from 'react-native'
import RoundedButton from './RoundedButton'

const BlueButton = (props) => (
  <TouchableOpacity style={styles.increaseButton}>
    <Text style={styles.buttonText} onPress={props.onPress}>{props.text}</Text>
  </TouchableOpacity>
)

const RedButton = (props) => (
  <TouchableOpacity style={styles.decreaseButton}>
    <Text style={styles.buttonText} onPress={props.onPress}>{props.text}</Text>
  </TouchableOpacity>
)

export default class Counter extends Component {
  // // Prop type warnings
  static propTypes = {
    value: PropTypes.number,
    onSyncIncrease: PropTypes.func,
    onSyncDecrease: PropTypes.func,
    onAsyncIncrease: PropTypes.func,
    onAsyncDecrease: PropTypes.func
  }
  //
  // Defaults for props
  static defaultProps = {
    value: 0
  }

  render = () => (
    <View style={styles.container}>
    
      <View style={styles.syncButton} >
        <BlueButton text='SYNC+' onPress={this.props.onSyncIncrease} />
        <RedButton text='SYNC-' onPress={this.props.onSyncDecrease} />
      </View>
  
      <Text style={styles.resultText}>{this.props.value}</Text>

      <View style={styles.asyncButton} >
        <BlueButton text='ASYN+' onPress={this.props.onAsyncIncrease} />
        <RedButton text='ASYN-' onPress={this.props.onAsyncDecrease} />
      </View>

    </View>
  )
}
