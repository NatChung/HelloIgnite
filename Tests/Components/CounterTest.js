import 'react-native'
import React from 'react'
import Counter from '../../App/Components/Counter'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'


test('Counter component renders correctly', () => {
    const tree = renderer.create(<Counter
        value={9}
        onSyncIncrease={() => { }}
        onSyncDecrease={() => { }}
        onAsyncIncrease={() => { }}
        onAsyncDecrease={() => { }}
    />).toJSON()
    expect(tree).toMatchSnapshot()
})

test('setValue', () => {
    let i = 0 // i guess i could have used sinon here too... less is more i guess
    const onPress = () => ++i
    const wrapperPress = shallow(<Counter value={13} />)

    expect(wrapperPress.find('Text').prop('children')).toBe(13)
})

test('onSyncIncrease', () => {
    let i = 0 // i guess i could have used sinon here too... less is more i guess
    const onPress = () => ++i
    const wrapperPress = shallow(<Counter onSyncIncrease={onPress} />)

    expect(wrapperPress.find('BlueButton').at(0).prop('onPress')).toBe(onPress) // uses the right handler
    expect(i).toBe(0)
    wrapperPress.find('BlueButton').at(0).simulate('press')
    expect(i).toBe(1)
})

test('onSyncDecrease', () => {
    let i = 0 // i guess i could have used sinon here too... less is more i guess
    const onPress = () => ++i
    const wrapperPress = shallow(<Counter onSyncDecrease={onPress} />)

    expect(wrapperPress.find('RedButton').at(0).prop('onPress')).toBe(onPress) // uses the right handler
    expect(i).toBe(0)
    wrapperPress.find('RedButton').at(0).simulate('press')
    expect(i).toBe(1)
})

test('onAsyncIncrease', () => {
    let i = 0 // i guess i could have used sinon here too... less is more i guess
    const onPress = () => ++i
    const wrapperPress = shallow(<Counter onAsyncIncrease={onPress} />)

    expect(wrapperPress.find('BlueButton').at(1).prop('onPress')).toBe(onPress) // uses the right handler
    expect(i).toBe(0)
    wrapperPress.find('BlueButton').at(1).simulate('press')
    expect(i).toBe(1)
})

test('onAsyncDecrease', () => {
    let i = 0 // i guess i could have used sinon here too... less is more i guess
    const onPress = () => ++i
    const wrapperPress = shallow(<Counter onAsyncDecrease={onPress} />)

    expect(wrapperPress.find('RedButton').at(1).prop('onPress')).toBe(onPress) // uses the right handler
    expect(i).toBe(0)
    wrapperPress.find('RedButton').at(1).simulate('press')
    expect(i).toBe(1)
})