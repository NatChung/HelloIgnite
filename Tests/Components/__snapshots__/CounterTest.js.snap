// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter component renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "row",
    }
  }
>
  <View
    style={
      Object {
        "width": 250,
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#3b5998",
          "borderRadius": 5,
          "height": 45,
          "justifyContent": "center",
          "margin": 15,
          "opacity": 1,
          "width": 100,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        onPress={[Function]}
        style={
          Object {
            "color": "white",
            "fontSize": 14,
            "fontWeight": "bold",
            "marginVertical": 10,
            "textAlign": "center",
          }
        }
      >
        SYNC+
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#e73536",
          "borderRadius": 5,
          "height": 45,
          "justifyContent": "center",
          "margin": 15,
          "opacity": 1,
          "width": 100,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        onPress={[Function]}
        style={
          Object {
            "color": "white",
            "fontSize": 14,
            "fontWeight": "bold",
            "marginVertical": 10,
            "textAlign": "center",
          }
        }
      >
        SYNC-
      </Text>
    </View>
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 60,
        "margin": 10,
        "marginVertical": 10,
        "textAlign": "center",
        "top": 20,
        "width": 230,
      }
    }
  >
    9
  </Text>
  <View
    style={
      Object {
        "width": 250,
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#3b5998",
          "borderRadius": 5,
          "height": 45,
          "justifyContent": "center",
          "margin": 15,
          "opacity": 1,
          "width": 100,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        onPress={[Function]}
        style={
          Object {
            "color": "white",
            "fontSize": 14,
            "fontWeight": "bold",
            "marginVertical": 10,
            "textAlign": "center",
          }
        }
      >
        ASYN+
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#e73536",
          "borderRadius": 5,
          "height": 45,
          "justifyContent": "center",
          "margin": 15,
          "opacity": 1,
          "width": 100,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        onPress={[Function]}
        style={
          Object {
            "color": "white",
            "fontSize": 14,
            "fontWeight": "bold",
            "marginVertical": 10,
            "textAlign": "center",
          }
        }
      >
        ASYN-
      </Text>
    </View>
  </View>
</View>
`;
